gemrb/core/System/String.cpp

```diff
@@ -100,6 +105,47 @@ static String* StringFromEncodedData(const ieByte* string, const EncodingStruct&
 	return dbString;
 }
 
+// returns new string converted to given encoding
+// in case iconv is not available, or there is encoding error, copy of original
+// string is returned
+const char* ConvertCharEncoding(IGNORE_UNUSED const char* string,
+        IGNORE_UNUSED const char* from, IGNORE_UNUSED const char* to) {
+#if HAVE_ICONV
+	iconv_t cd = iconv_open(to, from);
+	if (cd == (iconv_t)-1) {
+	    perror("iconv_open");
+	    return strdup(string);
+	}
+
+
+	char * in = (char *) string;
+	size_t in_len = strlen(string);
+	// twice as much bytes should be safe for most encodings
+	size_t out_len = in_len * 2 + 1;
+	size_t out_len_left = out_len;
+	char* buf = (char*) malloc(out_len);
+	char* buf_out = buf;
+	#if __FreeBSD__
+		size_t ret = iconv(cd, (const char **)&in, &in_len, &buf_out, &out_len_left);
+	#else
+		size_t ret = iconv(cd, &in, &in_len, &buf_out, &out_len_left);
+	#endif
+
+	if (ret == (size_t)-1) {
+	    perror("iconv");
+	    free(buf);
+	    return strdup(string);
+	}
+
+	size_t used = out_len - out_len_left;
+	buf = (char*)realloc(buf, used + 1);
+	buf[used] = '\0';
+	return buf;
+#else
+	return strdup(string);
```

Comment:
we should add a warning log message about no iconv I think.

Reply:
We could, but I'm afraid that would only obfuscate log, as this function can be called many times.